finesse/hardware/st10_controller.py

```diff
@@ -44,6 +175,13 @@ def __init__(self, serial: Serial) -> None:
             ST10ControllerError: Malformed message received from device
         """
         self.serial = serial
+        timeout = self.serial.timeout
+        self.serial.timeout = None
+
+        self._reader = _SerialReader(serial, timeout)
```

Comment:
Yeah, exactly. This bit of code probably needs a comment.

Setting the `timeout` to `None` means that the device will wait forever for a read, which is actually what we want as we need to be constantly reading from the device in case we get the magic async message. We can then emulate this functionality by only waiting for `timeout` seconds when we read from the `Queue` and raising an error if it fails.